### ISSUE TITLE
Limit GraphQL argument for `first`

### ISSUE THREAD

**Issue by yohix** [sourcegraph/sourcegraph#10127]
- **Sourcegraph version:** Sourcegraph.com

Limit GraphQL argument for `first`

#### Steps to reproduce:

1. Open GraphQL Explorer
2. Paste the following query

```
query {
  repositories(first: 10000000) {
    nodes {
      name
    }
  }
}
```
3. You will get **Error 504 Gateway time-out**

![Screenshot](https://user-images.githubusercontent.com/61746440/79991482-fe6eea80-84cf-11ea-8c9e-d5852fb11fa1.png)

**Comment by yohix** [comment]
@unknwon This seems like security issue as stated in https://www.apollographql.com/blog/securing-your-graphql-api-from-malicious-queries-16130a324a6b, I think need to add `perf` and `security` label

**Comment by unknwon** [comment]
Agree with `perf` label, don't know why it's a security problem?

**Comment by yohix** [comment]
> Agree with `perf` label, don't know why it's a security problem?

@unknwon It cause DOS in backend servers!

**Comment by unknwon** [comment]
I think that's because of https://github.com/sourcegraph/sourcegraph/issues/10129 not the "first" parameter alone?

**Comment by yohix** [comment]
Makes sense @unknwon this cause only performance issue not the security one :smile: 

PS: Consider pagination for this issue!
